src/thread/thread.cpp

```diff
@@ -113,8 +113,10 @@ namespace fc {
       if( my )
       {
         // wlog( "calling quit() on ${n}",("n",my->name) );
-        quit(); // deletes `my`
+        quit();
       }
+
+      delete my;
```

Comment:
If can delete `my` anyway, why need to check `if( my )` above?

Reply:
Sorry, I don't frankly get your point. It's perfectly safe in C++ to delete nullptr - it would do absolutely nothing. 

We absolutely need to check the my for null before calling quit() since the pointer would be dereferenced inside quit() thus causing a SIGSEGV.

Reply:
Oh, I didn't know `nullptr` can be safely deleted in C++ 11. My knowledge about `delete` is still on older version. Thanks.